_drafts/2020-08-07-g2p.md

```diff
@@ -0,0 +1,625 @@
+
+====
+***PROPOSED END OF SECOND POST***
+====
+
+## Applications
+
+There are number of different software tools that are already making use of `g2p`. For general purpose use in Python, have a look at [this section](#advanced-use-in-python). G2P is also used extensively in the [ReadAlongs](#readalongs) project for creating interactive audio/text documents and in [Convertextract](#convertextract), a tool for doing `g2p` transformations in Microsoft Office documents while preserving the original document formatting.
+
+### ReadAlongs
+
+ReadAlongs is an exploratory research project from the [National Research Council's Indigenous Language Technology Project](https://nrc.canada.ca/en/research-development/research-collaboration/programs/canadian-indigenous-languages-technology-project).
+
+Language revitalization communities have potentially many different recordings of language and some associated text, but mobilizing these materials into something that is educationally useful can be time consuming. It's also tough from the learner's perspective to try and follow along the text (maybe a Word Document) just by simply playing the audio. It's easy to get lost in the recording quickly, and from personal experience, trying to do that usually involves a lot of frustrating rewinding.
+
+What if we could develop a tool that *automatically* figured out what parts of the audio file corresponded to what parts of the text? In [Natural Language Processing](https://en.wikipedia.org/wiki/Natural_language_processing), this is called *forced alignment*, and this is fundamentally what the ReadAlongs project does. But how does it do it? In part, by using a whole bunch of `g2p` mappings!
+
+First of all, it's good to know that forced alignment is essentially a "solved problem" for languages with a lot of data, thanks to people like David Huggins-Daines who is a collaborator on the ReadAlongs project. And, if you know Python, and have 20+ hours of manually aligned text and audio in your language, you can train a model using one of the few tools out there developed for this task, like [the Montreal Forced Aligner](https://montreal-forced-aligner.readthedocs.io/en/latest/). But what if you don't have that much data (or any) and what if you don't know Python? If your language has a `g2p` mapping between its writing system and the ipa, you can likely circumvent that whole process. Here's how:
```

Comment:
Especially if ReadAlongs were its own blog post, going into the implications of having a trained model and what that means would be \*chefs kiss\*